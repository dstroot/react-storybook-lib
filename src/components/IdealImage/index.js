/**
 * This component uses a componenet to load and display images
 *
 * It requires external libraries:
 * `yarn add react-ideal-image react-waypoint`
 */

import React from 'react';
import IdealImage from 'react-ideal-image';
import image from './media/doggo.jpg';

// https://github.com/stereobooster/react-ideal-image/blob/master/introduction.md
const Image = () => (
  <IdealImage
    placeholder={{ color: '' }}
    srcSet={[{ src: image, width: 3500 }]}
    alt="doggo"
    width={3500}
    height={2095}
  />
);

export default Image;
