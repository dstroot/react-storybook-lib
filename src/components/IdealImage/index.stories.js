import React from 'react';
import { storiesOf } from '@storybook/react';

// component
import IdealImage from '../IdealImage';

// story
storiesOf('IdealImage', module).add('show IdealImage with test image', () => (
  <IdealImage />
));
