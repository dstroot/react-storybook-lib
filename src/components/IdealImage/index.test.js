import React from 'react';
import renderer from 'react-test-renderer';
import IdealImage from '../IdealImage';

describe('IdealImage', () => {
  it('it should render', () => {
    const component = renderer.create(<IdealImage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
