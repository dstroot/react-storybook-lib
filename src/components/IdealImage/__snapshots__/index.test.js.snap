// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdealImage it should render 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "position": "relative",
    }
  }
>
  <svg
    height={2095}
    style={
      Object {
        "height": "auto",
        "marginBottom": "-4px",
        "maxWidth": "100%",
        "width": "100%",
      }
    }
    width={3500}
  />
</div>
`;
